### PR TITLE
LPS-175996 Make triggerBy be 'Not Found' if notificationHandler is null

### PR DESCRIPTION
**Pull request by devpedrotavares**
Related issue: [LPS-175996](https://issues.liferay.com/browse/LPS-175996)

### CODE REVIEW

File: modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationQueueEntryResourceImpl.java [hunk 1]
```diff
@@ -177,8 +177,15 @@ private NotificationQueueEntry _toNotificationQueueEntry(
...
-					contextAcceptLanguage.getPreferredLocale());
+
+				if (notificationHandler != null) {
+					triggerBy = notificationHandler.getTriggerBy(
+						contextAcceptLanguage.getPreferredLocale());
+				}
+				else {
+					triggerBy = "Not Found";
+				}
+
```
Comment (carolmariaabb) [inline modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationQueueEntryResourceImpl.java:188]: @devpedrotavares Could you explain how this change fixes the described issue [LPS-175996](https://issues.liferay.com/browse/LPS-175996)? It is expected to return `Not Found`?

In addition, shouldn't you use the language key instead?  `_language.get(locale, "not-found")`, for example

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationQueueEntryResourceImpl.java (hunk 1 above)
Comment (devpedrotavares) [reply modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationQueueEntryResourceImpl.java]: Yes, @carolmariaabb. When the objectDefinition that triggered the notification entry is deleted, there will be no notificationHandler for that objectDefinition, which causes NullPointerException, returning 500 status code. That's why doing that null check solves the issue.
With this in mind, I see two alternatives: pass an empty string or pass some message denoting that the trigger object was not found (or does not exist).
If we opt to pass a message, yes we should use the language key as you said, not using it was a mistake.

**Comment by devpedrotavares** [comment]
ci:test:sf

File: modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationQueueEntryResourceImpl.java (hunk 1 above)
Comment (devpedrotavares) [reply modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationQueueEntryResourceImpl.java]: This is the output: 
not-found-result

Comment (carolmariaabb) [reply modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationQueueEntryResourceImpl.java]: @devpedrotavares please confirm this new behavior with the designers and Ana (or Luiz). Thanks!

**Comment by devpedrotavares** [comment]
ci:test:sf

**Comment by devpedrotavares** [comment]
ci:test:object

**Comment by devpedrotavares** [comment]
ci:test:relevant

File: modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationQueueEntryResourceImpl.java (hunk 1 above)
Comment (devpedrotavares) [reply modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationQueueEntryResourceImpl.java]: Just updated with the message as discussed in this [thread](https://liferay.slack.com/archives/C026UR5976G/p1679589546371649)

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 2]
```diff
@@ -10214,6 +10214,7 @@ miscellaneous=Miscellaneous
 missing-application-name=The application name is missing.
 missing-instance-error=Please select at least one instance before executing an Action.
 missing-manifest.xml=The file manifest.xml is not in the the LAR.
+missing-object-definition=Missing Object
```
Comment (carolmariaabb) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:10217]: @devpedrotavares shouldn't it be `Missing Object Definition`?

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties (hunk 2 above)
Comment (carolmariaabb) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:10217]: @devpedrotavares ok, let's wait for Brian's review then

**Review by carolmariaabb — APPROVED** [review]

**Pull request closed without merging**